stop_distance: 7
xdiff_threshold: 3